From ff3dcfed5fa83dc1be2fee44213c282af02127aa Mon Sep 17 00:00:00 2001
From: CanYang He <hcy@rock-chips.com>
Date: Mon, 23 Apr 2018 15:38:12 +0800
Subject: [PATCH] arm64: dts: rockchip: support lpddr4 for rk3399

1. modify lpddr4 setting in rk3399-dram-default-timing.dts
2. add a dts file for evb-rev3 lpddr4
3. using sram buffer for pcm buffer

Change-Id: Id52e9e4b5c366ec61bfb4818d3a14a4ea2024806
Signed-off-by: CanYang He <hcy@rock-chips.com>
Signed-off-by: Sugar Zhang <sugar.zhang@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/Makefile         |   1 +
 .../rockchip/rk3399-dram-default-timing.dtsi  |  12 +-
 .../rockchip/rk3399-evb-rev3-android-lp4.dts  | 171 ++++++++++++++++++
 3 files changed, 178 insertions(+), 6 deletions(-)
 create mode 100644 arch/arm64/boot/dts/rockchip/rk3399-evb-rev3-android-lp4.dts

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-dram-default-timing.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-dram-default-timing.dtsi
index 72cf3eb5b2b4..981777b82329 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-dram-default-timing.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-dram-default-timing.dtsi
@@ -68,13 +68,13 @@
 		phy_lpddr3_dq_drv = <PHY_DRV_ODT_34_3>;
 		phy_lpddr3_odt = <PHY_DRV_ODT_240>;
 
-		lpddr4_odt_dis_freq = <933>;
-		lpddr4_drv = <LP4_PDDS_60ohm>;
+		lpddr4_odt_dis_freq = <800>;
+		lpddr4_drv = <LP4_PDDS_240ohm>;
 		lpddr4_dq_odt = <LP4_DQ_ODT_40ohm>;
-		lpddr4_ca_odt = <LP4_CA_ODT_40ohm>;
+		lpddr4_ca_odt = <LP4_CA_ODT_DIS>;
 		phy_lpddr4_ca_drv = <PHY_DRV_ODT_40>;
-		phy_lpddr4_ck_cs_drv = <PHY_DRV_ODT_80>;
-		phy_lpddr4_dq_drv = <PHY_DRV_ODT_80>;
-		phy_lpddr4_odt = <PHY_DRV_ODT_60>;
+		phy_lpddr4_ck_cs_drv = <PHY_DRV_ODT_40>;
+		phy_lpddr4_dq_drv = <PHY_DRV_ODT_60>;
+		phy_lpddr4_odt = <PHY_DRV_ODT_40>;
 	};
 };
